test/unit/browsered/field_validation_checks_test.js

```diff
@@ -41,4 +47,85 @@ describe('field validation checks', () => {
       expect(isFieldGreaterThanMaxLengthChars('123456ABC', 'que')).to.equal(false)
     })
   })
+  describe('isNotValidAccountNumber', () => {
+    it('should validate successfully for 8 digits', () => {
+      const accountNumber = '00012345'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.false // eslint-disable-line
+    })
+
+    it('should validate successfully for 6 digits', () => {
+      const accountNumber = '012345'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.false // eslint-disable-line
+    })
+
+    it('should validate successfully for 7 digits', () => {
+      const accountNumber = '0012345'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.false // eslint-disable-line
+    })
+
+    it('should be not valid when is not a number', () => {
+      const accountNumber = 'abcdefgh'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.equal('Please enter a valid account number')
+    })
+
+    it('should be not valid when is too short', () => {
+      const accountNumber = '12345'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.equal('Please enter a valid account number')
+    })
+
+    it('should be not valid when is too long', () => {
+      const accountNumber = '123456789'
+
+      expect(isNotAccountNumber(accountNumber)).to.be.equal('Please enter a valid account number')
+    })
+  })
+
+  describe('isNotValidSortCode', () => {
+    it('should validate successfully for 6 digits', () => {
+      const sortCode = '108800'
+
+      expect(isNotSortCode(sortCode)).to.be.false // eslint-disable-line
+    })
+
+    it('should validate successfully for 6 digits with dashes', () => {
+      const sortCode = '10-88-00'
+
+      expect(isNotSortCode(sortCode)).to.be.false // eslint-disable-line
+    })
+
+    it('should validate successfully for 6 digits with spaces', () => {
+      const sortCode = '10 88 00'
+
+      expect(isNotSortCode(sortCode)).to.be.false // eslint-disable-line
+    })
+
+    it('should validate successfully for 6 digits with mix of dashes and spaces', () => {
+      const sortCode = '10-88 00'
+
+      expect(isNotSortCode(sortCode)).to.be.false // eslint-disable-line
+    })
```

Comment:
No tests for doing something like `1 0 88 00` even though our regex would appear to allow it — do we want that?

Reply:
We allow `10-88 00` case because we are going to do some normalisation, so we decided that `1 0 88 00` is valid as well and added the test, thanks